patchwork/views/__init__.py

```diff
@@ -273,48 +275,65 @@ def generic_list(
         else:
             context['filters'].set_status(filterclass, setting)
 
-    if patches is None:
-        patches = Patch.objects.filter(project=project)
+    if series_view:
```

Comment:
Given we're not modifying anything here, I don't think there's any reason to re-use this function. It's already way too complicated for it's own good. Better to split this out into a new view function, IMO.